src/ImageSharp/Common/Memory/BufferPointer.cs

```diff
@@ -87,5 +110,62 @@ public static int SizeOf<T>(int count)
         public static uint USizeOf<T>(int count)
             where T : struct
             => (uint)SizeOf<T>(count);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryMarshalCopy<TSource, TDest>(BufferPointer<TSource> source, BufferPointer<TDest> destination, int count)
+            where TSource : struct
+            where TDest : struct
+        {
+            // Pattern Based On:
+            // https://github.com/dotnet/corefx/blob/master/src/System.Numerics.Vectors/src/System/Numerics/Vector.cs#L12
+            //
+            // Note: The following patterns are used throughout the code here and are described here
+            //
+            // PATTERN:
+            //    if (typeof(T) == typeof(Int32)) { ... }
+            //    else if (typeof(T) == typeof(Single)) { ... }
+            // EXPLANATION:
+            //    At runtime, each instantiation of BufferPointer<T> will be type-specific, and each of these typeof blocks will be eliminated,
+            //    as typeof(T) is a (JIT) compile-time constant for each instantiation. This design was chosen to eliminate any overhead from
+            //    delegates and other patterns.
+            if (typeof(TSource) == typeof(long))
```

Comment:
I remember thinking I was going to have to do something like this before I discovered the unsafe type. Interesting to see it come back.